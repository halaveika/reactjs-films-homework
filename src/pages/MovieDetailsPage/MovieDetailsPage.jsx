import React, { useState } from 'react';
import { Layout } from 'antd';
import InfoBar from '../../components/InfoBar';
import NavBar from '../../components/NavBar';
import { BACKDROP_PATH_URL } from '../../constants';
import './MovieDetailsPage.scss';

export default function MovieDetailsPage({id, title, genres, vote_average, poster, overview, runtime, video, GetVideoUrl }) {

  return (
    <Layout className="movieDetails-container">
      <img className="movieDetails-image" src={`${BACKDROP_PATH_URL}${poster}`} alt="wallpaper" />

      <Layout.Content className="movieDetails-content">
        <InfoBar
          title={title}
          genres={genres}
          vote_average={vote_average}
          runtime={runtime}
        />
        <NavBar
          overview={overview}
          video={video}
          GetVideoUrl={GetVideoUrl}
          id={id}
        />
      </Layout.Content>
    </Layout>
  );
}
