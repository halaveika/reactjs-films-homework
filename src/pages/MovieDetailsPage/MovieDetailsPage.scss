.movieDetails {
  position: relative;
  width: 100vw;
  height: 100vh;
  background-color: transparent;
}

.movieDetails-image {
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: -1;
}

.movieDetails-container {
  margin: 0 auto;
  padding: 30px 75px 30px 75px;
  max-width: 1200px;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-end;
  background-color: transparent;
}
