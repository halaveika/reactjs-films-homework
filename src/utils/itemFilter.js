const itemfilter = (item, genre) => item
  .filter((e) => ((genre) ? e.genres.filter((name) => name === genre).length > 0 : e));

export default itemfilter;
