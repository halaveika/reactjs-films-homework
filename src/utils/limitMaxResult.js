import { MAX_SEARCH_RESULT, MAX_PAGE } from '../constants';

const limitMaxResult = (result) => {
  if (result.total_pages > MAX_PAGE) { result.total_pages = MAX_PAGE; }
  if (result.total_results > MAX_SEARCH_RESULT) { result.total_results = MAX_SEARCH_RESULT; }
  return result;
};

export default limitMaxResult;
