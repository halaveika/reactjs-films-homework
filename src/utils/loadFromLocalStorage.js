const loadFromLocalStorage = () => {
  const serializedState = localStorage.getItem('state');
  if (serializedState === null) return undefined;
  return JSON.parse(serializedState);
};

export default loadFromLocalStorage;
