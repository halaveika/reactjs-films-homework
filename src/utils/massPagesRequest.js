import {API_PAGE_SIZE} from '../constants';

const massPagesRequest = async(page, pageSize, httpService, str='') => {
  console.log('page: ' + page);
  console.log('pageSize: ' + pageSize);
  console.log('str: ' + str);
  const countPages = Math.ceil(pageSize / API_PAGE_SIZE);
  console.log('countPages: ' + countPages);
  let json;
  if(str){
    json = await Promise.all(Array(countPages).fill(1).map(async(_, index) =>  await httpService(str, page + index)));
  } else {
    json = await Promise.all(Array(countPages).fill(1).map(async(_, index) =>  await httpService(page + index)));
  }
  return {
    page: page,
    results: json.reduce((arr,e) => arr = arr.concat(e.results),[]),
    total_pages: json[json.length - 1].total_pages,
    total_results: json[json.length - 1].total_results,
  }

}

export default massPagesRequest;
