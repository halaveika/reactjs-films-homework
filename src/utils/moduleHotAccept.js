module.exports = {
  moduleHotAccept(mod) {
    if (mod && mod.hot) {
      mod.hot.accept();
    }
  },
};
