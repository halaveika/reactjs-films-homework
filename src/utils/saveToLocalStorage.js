const saveToLocalStorage = (state) =>{
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  }

  export default saveToLocalStorage
