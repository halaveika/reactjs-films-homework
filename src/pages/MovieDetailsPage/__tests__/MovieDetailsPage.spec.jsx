import React from 'react';
import { create, act } from 'react-test-renderer';
import '../../../utils/matchMedia';
import MovieDetailsPage from '..';

describe('test MovieDetailsPage component', () => {
  let component;
  act(() => { component = create(<MovieDetailsPage />); });
  const tree = component.toJSON();
  it('should render MovieDetailsPage component', () => {
    expect(tree).toMatchSnapshot();
  });
});
