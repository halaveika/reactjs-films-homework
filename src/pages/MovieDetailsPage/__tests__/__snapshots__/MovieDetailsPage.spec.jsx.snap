// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MovieDetailsPage component should render MovieDetailsPage component 1`] = `
<Layout
  className="movieDetails"
>
  <img
    alt="wallpaper"
    className="movieDetails-image"
    src="assets/images/cover-image.jpg"
  />
  <Content
    className="movieDetails-container"
  >
    <InfoBar
      genres={
        Array [
          "drama",
          "sport",
          "comedy",
        ]
      }
      runtime={120}
      title="Fight club"
      vote_average={8}
    />
    <NavBar
      id={2412412}
      overview="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry"
    />
  </Content>
</Layout>
`;
